color.d

```diff
@@ -1777,14 +1779,45 @@ struct Point {
 	Point opBinary(string op)(int rhs) @nogc {
 		return Point(mixin("x" ~ op ~ "rhs"), mixin("y" ~ op ~ "rhs"));
 	}
+
+	Size opCast(T = Size)() inout @nogc {
+		return Size(x, y);
+	}
 }
 
 ///
 struct Size {
 	int width; ///
 	int height; ///
 
-	int area() pure nothrow @safe const @nogc { return width * height; }
+	pure nothrow @safe:
+
+	int area() const @nogc { return width * height; }
+
+	int offset(const int x, const int y) const @nogc {
+		return linearOffset(width, Point(x, y));
+	}
+
+	int offset(const Point p) const @nogc {
+		return linearOffset(width, p);
+	}
+
+	Point opCast(T = Point)() inout @nogc {
+		return Point(width, height);
+	}
+}
+
+/++
+	Calculates the linear offset of a point
+	from the start (0/0) of a rectangle
+ +/
+int linearOffset(const Point pos, const int width) @safe pure nothrow @nogc {
```

Comment:
This does assume that the offset works a certain way. The implementation of course makes it clear what it does but the doc should prolly explain it too (even if it just literally says `Returns: y*width + x` cuz that's such a recognizable pattern anyway). But like think of a Windows bmp which traditionally puts (0, 0) in the lower-left corner of the image, and thus the memory offset is "upside down" compared to the screen drawing, which puts (0,0) in the upper left.

Reply:
Is the new explanation sufficient?